internal/hook/item_hook_handler.go

```diff
@@ -239,3 +317,61 @@ func (r ResourceHook) applicableTo(groupResource schema.GroupResource, namespace
 	}
 	return true
 }
+
+// ResourceRestoreHook is a restore hook for a given resource.
+type ResourceRestoreHook struct {
+	Name         string
+	Selector     ResourceHookSelector
+	RestoreHooks []velerov1api.RestoreResourceHook
+}
+
+func getInitRestoreHookFromAnnotation(annotations map[string]string) *velerov1api.InitRestoreHook {
+	containerImage := annotations[podRestoreHookInitContainerImageAnnotationKey]
+	containerName := annotations[podRestoreHookInitContainerNameAnnotationKey]
+	command := annotations[podRestoreHookInitContainerCommandAnnotationKey]
+	if command == "" || containerImage == "" || containerName == "" {
```

Comment:
I wonder whether users would expect to be able to just supply an image and default to ENTRYPOINT for command and generated name for name? If returning from here and at least one of these is non-empty would it make sense to log?

Reply:
Yes, I think we'll want to allow users to use the default `ENTRYPOINT` on the images.
I'll clean that up. Along with fixing the CRD which has an unused kubebuilder constraint.
Adding a log message would also help.

Reply:
I think that's a good idea, too. Is that consistent with how backup hooks work now? If they don't support it, we should make an issue to get that in at some point, but not for 1.5.

Reply:
I think the backup hook case is slightly different where handling the hook is running a command inside of a container. Where it is assumed that the container already exists on the pod and updating the pod spec to add a new one is not necessary. For that reason, this will be something specific to the restore hook scenario.

Reply:
@ashish-amarnath Got it, that makes sense. Thanks!